packages/block-library/src/blocks/header/style.scss

```diff
@@ -1,61 +1,230 @@
 @import 'scss/menu-toggle';
 
 @media only screen and (min-width: 1000px) {
-	// Nova Blocks hacks
-	.site-header {
-		// styles from Nova Blocks should only affect blocks created by it
-		@import 'scss/menu';
-	}
-
-	.site-header__content {
-		display: grid;
-		align-items: center;
-		grid-gap: 1em;
-
-		> * {
-			justify-content: center;
-		}
-
-		> :first-child {
-			justify-content: flex-start;
-		}
-
-		> :last-child {
-			justify-content: flex-end;
-		}
-
-		> .wp-block-novablocks-navigation {
-			> .menu,
-			> .menu > ul {
-				justify-content: inherit;
-			}
-		}
-
-		.site-header--logo-left & {
-			grid-template-columns: auto 1fr;
-		}
-
-		.site-header--logo-center & {
-			grid-template-columns: 1fr auto 1fr;
-		}
-
-		.site-header--logo-right & {
-			grid-template-columns: 1fr auto;
-		}
-
-		&.alignfull[class] {
-			margin-left: 0;
-			margin-right: 0;
-		}
-	}
-
-	.c-menu-toggle__checkbox,
-	.c-menu-toggle {
-		display: none;
-	}
+  // Nova Blocks hacks
+  .site-header,
+  .site-header-sticky {
+    // styles from Nova Blocks should only affect blocks created by it
+    @import 'scss/menu';
+  }
+  
+  .site-header__content {
+    
+    > .wp-block-novablocks-navigation {
+      > .menu,
+      > .menu > ul {
+        justify-content: inherit;
+      }
+    }
+    
+    &.alignfull[class] {
+      margin-left: 0;
+      margin-right: 0;
+    }
+  }
+  
+  .c-menu-toggle__checkbox,
+  .c-menu-toggle {
+    display: none;
+  }
+  
+  // Fix Header layout until
+  // user hit save after migration.
+ .site-header--is-old {
+   
+   &.site-header--logo-left {
+     
+     .site-header__content {
+       grid-template-columns: auto 1fr;
+     }
+   }
+    .site-header__content {
+      display: grid;
+      grid-template-columns: 1fr auto 1fr;
+      
+      > :last-child {
+        justify-content: flex-end;
+      }
+    }
+  }
 }
 
 @media not screen and (min-width: 1000px) {
-	@import 'scss/mobile';
-	@import 'scss/mobile-transitions';
+  @import 'scss/mobile';
+  @import 'scss/mobile-transitions';
+}
+
+.site-header-sticky {
+  position: fixed;
+  top: var(--admin-bar-height, 0);
+  
+  width: 100%;
+  z-index: 1100;
+  
+  opacity: 0;
+  pointer-events: none;
+  
+  transition: opacity .15s ease-in;
+  
+  &--is-visible {
+    opacity: 1;
+    pointer-events: auto;
+  }
+}
+
+.site-header {
+  
+  &--is-sticky {
+    position: fixed;
+    top: var(--admin-bar-height, 0);
+  }
+}
+
+@include below(lap) {
+  
+  .site-header-sticky {
+    display: none;
+  }
+  
+  .site-header--simple {
+    
+    .site-header__row {
+      display: flex;
+      flex-direction: column;
+      
+      .site-header__menu--primary {
+        order: -1;
+      }
+    }
+  }
+  
+  .site-header__row--primary {
+    order: -1;
+  }
+}
+
+@include above(lap) {
+  
+  .site-header__row {
+    
+    display: flex;
+    align-items: center;
+    
+    &--secondary {
```

Comment:
Styling could be done based on the block-type and whether or not it is the first / last block inside the header row. I suggest trying this approach and only after that styling specific cases